[Feature] 간단 Unit Test 추가

Choe-ju opened the pull request:
## AS-IS
- 네트워크 호출, UI 상태 변경 등에 대해 수동으로 테스트
## TO-BE
- Unit Test를 통해 주요 비즈니스 로직 및 네트워크 호출 로직 자동 검증
## KEY-POINT
- 간단한 네트워크 응답테스트 추가
## SCREENSHOT (Optional)

Choe-ju committed:
feat: 시범 적용 용도 Unit Test 추가

Choe-ju committed:
feat: RecordSkeletonView 구현

Choe-ju committed:
fix: 각 view 애니메이션 효과 누적 방지

kwon-seokki commented on HongikYeolgong2Tests/NetworkServiceTests.swift:
```diff
@@ -0,0 +1,54 @@
+        // KeyChainManager accessToken 세팅
+        key = KeyChainManager.readItem(key: .accessToken) ?? ""
+        KeyChainManager.deleteItem(key: .accessToken)
+        KeyChainManager.addItem(key: .accessToken, value: testToken)
```
키체인에 민감한 정보를 저장하다 보니깐 이값을 조정하기 보다는 다른 방법이 필요해 보입니다!

kwon-seokki commented on HongikYeolgong2Tests/NetworkServiceTests.swift:
```diff
@@ -0,0 +1,54 @@
+@testable import HongikYeolgong2
+
+final class NetworkServiceTests: XCTestCase {
+    var networkService: NetworkService!
```
테스트 대상 unit은 주로 SUT(System Under Test) 라고 네이밍 한다고 하네요?
네이밍에 신경 쓰지않고 대상unit은 앞으로 sut로 통일해도 좋을듯 합니다!

Choe-ju replied:
넵 수정해주겠습니다

kwon-seokki commented on HongikYeolgong2/Data/DTO/StudySession/StudyTimeResponseDTO.swift:
```diff
@@ -7,7 +7,7 @@
 import Foundation
 
-struct StudyTimeResponseDTO: Decodable {
+struct StudyTimeResponseDTO: Codable {
```
해당 객체가 인코딩이 필요한 경우가 있나요?

Choe-ju replied:
이전에 값 불러와서 비교 테스트할때 필요했었는데 지금은 필요 없어졌는데 복구시켜 놓는다는 게 깜빡했네요
다시 수정해두겠습니다

kwon-seokki reviewed (commented):
코멘트 남겼습니다 한번씩 확인해주세요!

Choe-ju committed:
fix: 코멘트 부분 보완사항 수정

owllwo98 commented:
저도 확인완료했습니다!

Choe-ju committed:
fix: NetworkServiceTests 토큰 접근 방식 개선

- 기존 KeyChainManager를 통해 토큰 값을 불러오는 방식에서
  테스트용 토큰을 동적으로 주입하는 방식으로 변경
  - NetworkService에 `testToken` 클로저 추가
  - URLRequest 확장으로 테스트 토큰 주입 가능하도록 수정

kwon-seokki commented on HongikYeolgong2/Presentation/Record/RecordSkeletonView.swift:
```diff
@@ -0,0 +1,82 @@
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            // 상단 시간 및 날짜
+            HStack {
+                RoundedRectangle(cornerRadius: 10)
+                    .fill(skeletonColor.opacity(opacity))
+                    .frame(width: 110.adjustToScreenWidth, height: 30.adjustToScreenHeight)
+                
+                Spacer()
+            }
+            
+            Spacer().frame(height: 12.adjustToScreenHeight)
+            
+            // 요일
+            HStack(alignment: .center) {
+                ForEach(0..<7, id: \.self) { _ in
+                    Spacer()
+                    RoundedRectangle(cornerRadius: 5)
+                        .fill(skeletonColor.opacity(opacity))
+                        .frame(maxWidth: .infinity, maxHeight: 17)
+                    Spacer()
+                }
+            }
+            Spacer().frame(height: 8.adjustToScreenHeight)
+            
+            // 달력
+            LazyVGrid(columns: Array(repeating: GridItem(.flexible(), spacing: 5.adjustToScreenWidth), count: 7), spacing: 5.adjustToScreenHeight) {
+                ForEach(0..<35, id: \.self) { _ in
+                    RoundedRectangle(cornerRadius: 8)
+                        .fill(skeletonColor.opacity(opacity))
+                        .frame(width: 40.adjustToScreenWidth,height: 33.adjustToScreenHeight)
+                }
+            }
+            
+            Spacer()
+            
+            // 하단 통계
+            VStack(spacing: 13.adjustToScreenWidth) {
+                ForEach(0..<2, id: \.self) { _ in
```
스켈레톤은 사용하지 않을듯합니다!!

kwon-seokki commented on HongikYeolgong2/Util/API/Base/NetworkService.swift:
```diff
@@ -3,15 +3,17 @@ import Foundation
     }()
     
-    init(session: URLSessionProtocol) {
+    init(session: URLSessionProtocol, testToken: @escaping () -> String? = {nil}) {
```
혹시 토큰을 클로저로 받는이유가 있을까요?
